xDrip/Components/Charts/SupportingLogic/BasalChart/BasalChartDataWorker.swift

```diff
@@ -13,20 +13,25 @@ enum BasalChartDataWorker {
     static func fetchBasalData(for hours: Int) -> [InsulinEntry] {
         let minimumDate = Date() - TimeInterval(hours) * .secondsPerHour
         let all = InsulinEntriesWorker.fetchAllBasalEntries()
-        return all.filter { $0.date >=? minimumDate }
+        return all.filter { $0.date >=? minimumDate && $0.cloudUploadStatus != .waitingForDeletion }
```

Comment:
Instead of checking for cloud upload status, let's add an `isValid: Bool` property to that entity, something like:
```
var isValid: Bool {
  return cloudUploadStatus != .waitingForDeletion
}
```

In this case we will be able to add other conditions in the future